projects/01-rubygems-cli/solution/lib/api/ruby_gems_api.rb

```diff
@@ -1,33 +1,35 @@
 require "faraday"
 require "faraday/net_http"
 require "json"
-require "./errors/gem_not_found_error"
-require "./errors/standard_api_error"
+require "./lib/errors/gem_not_found_error"
+require "./lib/errors/standard_api_error"
+require "./lib/api/validator"
 
-class RubyGemsApi
+class RubyGemsAPI
   @connection = Faraday.new("https://rubygems.org") do |faraday|
     faraday.adapter :net_http
+    faraday.request :authorization, "Bearer", ENV['RUBYGEMS_API_KEY']
   end
 
   class << self
-    def gem_info(gem_name)
-      raise ArgumentError.new("Invalid gem name provided.") if gem_name.nil? || gem_name.empty?
-      
+    def show_gem(gem_name)
+      raise ArgumentError.new("Invalid gem name provided.") if !Validator.gem_name_valid?(gem_name)
```

Comment:
You can go positive all the way:
```ruby
if Validator.gem_name_invalid?(gem_name)
```
or maybe do a generic `invalid?` method, e.g. `Validator.invalid?(gem_name)`, since it reads out pretty clearly & that method could encapsulate whatever logic needed to consider the input as invalid.